### DISCUSSION TITLE
stack panel won't center

### DISCUSSION THREAD

**Discussion by CosminCAS123** [AvaloniaUI/Avalonia#13833] [General]
I have a  stack panel inside of another stack panel. I want to horizontally center the child one, but HorizontalAlignment won't do it:
![image](https://github.com/AvaloniaUI/Avalonia/assets/79158479/5e052319-7ad6-40f3-9f62-9a4b154c1c72)
Any idea?

**Comment by thevortexcloud** [comment]
What does dev tools show (Press F12 while the app is running)?

**Comment by timunie** [comment]
The outer StackPanel is horizontal and will grow infinite acc. to children count. So you cannot center anything here. Use another Panel like Grid or DockPanel for example.

**Reply by maxkatz6** [reply]
Yes, it doesn't make much sense centering control inside of a StackPanel remaining space. 
Wanted behavior can be achieved with WrapPanel though, which gives whole remaining space to the last control, so it can be centered.
